chatbot-core/tests/unit/routes/test_file_upload.py

```diff
@@ -0,0 +1,168 @@
+    mock_get_chatbot_reply.return_value = {"reply": "I see an image."}
+    
+    # Create a mock image file (PNG header)
+    file_content = b"\x89PNG\r\n\x1a\n" + b"\x00" * 100
+    files = [
+        ("files", ("screenshot.png", BytesIO(file_content), "image/png"))
+    ]
+    
+    response = client.post(
+        "/sessions/test-session-id/message/upload",
+        data={"message": "What's in this image?"},
+        files=files
+    )
+    
+    assert response.status_code == 200
+    data = response.json()
+    assert "reply" in data
+
+
+def test_chatbot_reply_with_multiple_files(client, mock_session_exists, mock_get_chatbot_reply):
+    """Test POST /sessions/{session_id}/message/upload with multiple files."""
+    mock_session_exists.return_value = True
+    mock_get_chatbot_reply.return_value = {"reply": "I analyzed the files."}
+    
+    files = [
+        ("files", ("file1.txt", BytesIO(b"Content 1"), "text/plain")),
+        ("files", ("file2.log", BytesIO(b"Content 2"), "text/plain")),
+    ]
+    
+    response = client.post(
+        "/sessions/test-session-id/message/upload",
+        data={"message": "Analyze these logs."},
+        files=files
+    )
+    
+    assert response.status_code == 200
+
+
+def test_chatbot_reply_upload_invalid_session(client, mock_session_exists):
+    """Test that upload endpoint returns 404 for invalid session."""
+    mock_session_exists.return_value = False
+    
+    files = [
+        ("files", ("test.txt", BytesIO(b"content"), "text/plain"))
+    ]
+    
+    response = client.post(
+        "/sessions/invalid-session/message/upload",
+        data={"message": "Test message"},
+        files=files
+    )
+    
+    assert response.status_code == 404
+    assert response.json()["detail"] == "Session not found."
+
+
+def test_chatbot_reply_upload_unsupported_file_type(client, mock_session_exists):
+    """Test that upload endpoint rejects unsupported file types."""
+    mock_session_exists.return_value = True
+    
+    files = [
+        ("files", ("archive.zip", BytesIO(b"PK..."), "application/zip"))
+    ]
+    
+    response = client.post(
+        "/sessions/test-session-id/message/upload",
+        data={"message": "Extract this archive."},
+        files=files
+    )
+    
+    assert response.status_code == 400
+    assert "Unsupported file type" in response.json()["detail"]
+
+
+def test_chatbot_reply_upload_empty_message_with_files(client, mock_session_exists, mock_get_chatbot_reply):
+    """Test that upload endpoint works with empty message if files provided."""
+    mock_session_exists.return_value = True
+    mock_get_chatbot_reply.return_value = {"reply": "I analyzed the file."}
+    
+    files = [
+        ("files", ("test.txt", BytesIO(b"Content"), "text/plain"))
+    ]
+    
+    response = client.post(
+        "/sessions/test-session-id/message/upload",
+        data={"message": ""},
+        files=files
+    )
+    
+    assert response.status_code == 200
+
```

Comment:
Verify that the "Truncation" logic in file_service.py works when called through the API. by creating `def test_chatbot_reply_text_truncation(client, mock_session_exists, mock_get_chatbot_reply):`